bitchat/Services/BluetoothMeshService.swift

```diff
@@ -2898,7 +2909,9 @@ class BluetoothMeshService: NSObject {
         case .protocolNack:
             // Handle protocol-level negative acknowledgment
             let senderID = packet.senderID.hexEncodedString()
-            if !isPeerIDOurs(senderID) {
+            if let recipientIDData = packet.recipientID,
+               isPeerIDOurs(recipientIDData.hexEncodedString())
+               && !isPeerIDOurs(senderID) {
```

Comment:
Related: we (Alice) react to NACKs from Bob, even if they are addressed to Carol and tear down the session.